kernel/src/thread/task.rs

```diff
@@ -88,15 +90,29 @@ pub fn create_new_user_task(
             if current_thread.is_exited() {
                 break;
             }
-            handle_pending_signal(user_ctx, &ctx, syscall_number);
-            // If current is suspended, wait for a signal to wake up self
-            while current_thread.is_stopped() {
-                Thread::yield_now();
-                debug!("{} is suspended.", current_posix_thread.tid());
-                handle_pending_signal(user_ctx, &ctx, None);
+
+            loop {
+                handle_pending_signal(user_ctx, &ctx, syscall_number);
```

Comment:
Can you confirm that handling multiple nested signals, e.g., SIGSTOP, SIGINT, SIGTRAP, and SIGCONT, is fine here? At least, our implementation should be compatible with it. (I'm not sure about Linux's behavior, but we can study it in the future.)

Reply:
OK. Thanks for the explanation.

I'm surprised to learn that `handle_pending_signal` will process signals with user handlers immediately instead of waiting for the next user loop.

This does not sound right to me. However, this is a separate issue that we may want to address in the future.